x-pack/plugins/security_solution/server/endpoint/services/actions/create/index.ts

```diff
@@ -130,19 +154,22 @@ export class ActionCreateService {
     // and allow only users with superuser privileges to write to fleet indices
     // const logger = endpointContext.logFactory.get('host-isolation');
     const doesLogsEndpointActionsDsExist = await doLogsEndpointActionDsExists({
-      esClient: this.esClient,
+      esClient,
       logger,
       dataStreamName: ENDPOINT_ACTIONS_DS,
     });
 
     // if the new endpoint indices/data streams exists
     // write the action request to the new endpoint index
     if (doesLogsEndpointActionsDsExist) {
-      logsEndpointActionsResult = await this.esClient.index<LogsEndpointAction>(
+      logsEndpointActionsResult = await esClient.index<LogsEndpointAction>(
         {
           index: ENDPOINT_ACTIONS_INDEX,
           body: {
             ...doc,
+            agent: {
```

Comment:
on second thought I think this change is not needed because the value is in `doc`. I'll adjust this soon.